runtime/lua/man.lua

```diff
@@ -8,7 +8,20 @@ local M = {}
 --- @param env? table<string,string|number>
 --- @return string
 local function system(cmd, silent, env)
-  local r = vim.system(cmd, { env = env, timeout = 10000 }):wait()
+  local ok, proc_or_err = pcall(vim.system, cmd, { env = env, timeout = 10000 })
+
+  if not ok then
+    local errmsg = tostring(proc_or_err)
+    if string.find(errmsg, 'ENOENT') then
+      error(string.format("can't find '%s'", cmd[1]), 0)
+    elseif string.find(errmsg, 'EACCES') then
```

Comment:
This is a bit clumsy. Maybe `vim.system()` should return a structured object or something, but meanwhile we can use `vim.fn.executable()` to do the check, without string-scraping error messages. Here's some untested sample code:

```lua
if vim.fn.executable(cmd[1]) == 0 then
  error(string.format('not executable (or does not exist): "%s"', cmd[1]), 0)
end
local r = vim.system(cmd, { env = env, timeout = 10000 }):wait()
```